pkg/controller/v1alpha2/applicationconfiguration/component.go

```diff
@@ -173,6 +186,17 @@ func (c *ComponentHandler) createControllerRevision(mt metav1.Object, obj runtim
 		c.l.Info(fmt.Sprintf("error create controllerRevision %v", err), "componentName", mt.GetName())
 		return false
 	}
+
+	// clean revisions
+	if curComp.Spec.RevisionLimit == nil {
+		curComp.Spec.RevisionLimit = &defaultRevisionLimit
+	}
+	if int64(*curComp.Spec.RevisionLimit) < nextRevision {
+		if err := c.cleanupControllerRevision(mt, curComp); err != nil {
```

Comment:
I assume we can create a separate goroutine to do this garbage collection thing, then it won't block the main logic.

Reply:
cleanupControllerRevision may be time consuming function

Reply:
Goroutine make the process async, that may create conflict， but we can use cache to optimize

Reply:
I'll create another PR changing all restful apis to informer machine, and that would modify many places.

Reply:
What makes conflict?

Reply:
I'm worry about clean revision that used when goroutine delayed and new reconcile triggered.  I'll change it

Reply:
That won't happen as we have  10 revisions as buffer.

Reply:
👌